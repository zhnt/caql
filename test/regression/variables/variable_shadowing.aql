// Variable shadowing test
let x = 10
let y = 20
print("outer: x=" + x + ", y=" + y)
{
    let x = 100  // shadows outer x
    let z = x + y  // uses inner x and outer y
    print("inner: x=" + x + ", y=" + y + ", z=" + z)
    {
        let y = 200  // shadows outer y
        let w = x + y  // uses inner x and inner y
        print("deepest: x=" + x + ", y=" + y + ", w=" + w)
    }
    print("back to inner: x=" + x + ", y=" + y)
}
print("back to outer: x=" + x + ", y=" + y)
