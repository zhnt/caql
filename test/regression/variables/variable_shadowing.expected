outer: x=10, y=20
inner: x=100, y=20, z=120
deepest: x=100, y=200, w=300
back to inner: x=100, y=20
back to outer: x=10, y=20

